### PR TITLE
Reduces code used to implement version

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Signed-off-by: Adrian Cole <adrian@tetrate.io>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -30,7 +30,7 @@ COVERAGE_REPORT := $(COVERAGE_DIR)/coverage.html
 GOOS := $(shell go env GOOS)
 GOARCH := $(shell go env GOARCH)
 
-GO_LD_FLAGS := -ldflags="-s -w -X github.com/tetratelabs/getenvoy/pkg/version.version=$(GETENVOY_TAG)"
+GO_LD_FLAGS := -ldflags="-s -w -X github.com/tetratelabs/getenvoy/internal/version.Current=$(GETENVOY_TAG)"
```
Comment (codefromthecrypt) [inline Makefile:33]: @llinder FYI: this can override variables at build time, but not const (var, not const)

File: pkg/cmd/app.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ func NewApp(o *globals.GlobalOpts) *cobra.Command {
 		DisableAutoGenTag: true, // removes autogenerate on ___ from produced docs
 		Short:             "Fetch and run Envoy",
 		Long:              `Manage Envoy lifecycle including fetching binaries and collection of process state.`,
-		Version:           version.Build.Version, // TODO: Add version support on the command
```
Comment (codefromthecrypt) [inline pkg/cmd/app.go:42]: old comment as -v is by default there (and how I tested this)

```bash
$ make bin GETENVOY_TAG=asdas
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags="-s -w -X github.com/tetratelabs/getenvoy/internal/version.Current=asdas" -o build/bin/linux/amd64/getenvoy ./cmd/getenvoy/main.go
CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 go build -ldflags="-s -w -X github.com/tetratelabs/getenvoy/internal/version.Current=asdas" -o build/bin/darwin/amd64/getenvoy ./cmd/getenvoy/main.go

$ build/bin/darwin/amd64/getenvoy -v
getenvoy version asdas
```

File: internal/version/version.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2021 Tetrate
```
Comment (mathetake) [inline internal/version/version.go:1]: so now what's the criteria for deciding whether a package should sit in interna/ or pkg/?

Comment (codefromthecrypt) [reply internal/version/version.go:1]: over time, everything except main.go will be internal, as we don't have a public api

**Comment by codefromthecrypt** [comment]
will rebase after #224 so we don't need to update two things with the same value

File: .goreleaser.yaml [hunk 4]
```diff
@@ -15,7 +15,7 @@
 project_name: getenvoy
 builds:
 - binary: getenvoy
-  ldflags: "-s -w -X github.com/tetratelabs/getenvoy/pkg/version.version={{.Version}}"
+  ldflags: "-s -w -X github.com/tetratelabs/getenvoy/internal/version.Current={{.Version}}"
```
Comment (codefromthecrypt) [inline .goreleaser.yaml:18]: only one time now!

**Pull request merged**